perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/PerunBlImpl.java

```diff
@@ -202,7 +202,10 @@ public void setUserExtSourceAttributes(PerunSession session, UserExtSource ues,
 
 					//for Array list attributes try to parse string value into individual fields
 					if(attributeWithValue.getType().equals(ArrayList.class.getName()) || attributeWithValue.getType().equals(BeansUtils.largeArrayListClassName)) {
-						List<String> value = new ArrayList<>(Arrays.asList(attrValue.split(UsersManagerBlImpl.multivalueAttributeSeparatorRegExp)));
+						List<String> value = new ArrayList<>();
+						if (attrValue != null) {
+							value = new ArrayList<>(Arrays.asList(attrValue.split(UsersManagerBlImpl.multivalueAttributeSeparatorRegExp)));
+						}
 						attributeWithValue.setValue(value);
 					} else {
 						attributeWithValue.setValue(attrValue);
```

Comment:
What about to rewrite this code as:
```
if(attrValue != null && (attributeWithValue.getType().equals(ArrayList.class.getName()) || attributeWithValue.getType().equals(BeansUtils.largeArrayListClassName))) {
  attrValue = new ArrayList<>(Arrays.asList(attrValue.split(UsersManagerBlImpl.multivalueAttributeSeparatorRegExp)));
}
attributeWithValue.setValue(attrValue);
```
Cause we are always setting attrValue no matter if it is null or splitted.

Reply:
I will keep it, since like this there would be conflicting type in attrValue variable.